### DISCUSSION TITLE
Inject resource info via processor in collector running as agent

### DISCUSSION THREAD

**Discussion by adriangb** [open-telemetry/opentelemetry-collector#3670] [Q&A] [answered]
Hi, how / what processor can I use to create a resource and inject it to every trace if I already have things like service name, env, etc. available in my Deployment metadata? I can't figure out if this is something k8s_processor, resourcedetection or another processor is supposed to do, or if I invariably have to do it at the application level using the SDKs.

**Comment by jrcamp** [comment, marked as answer]
Sounds like you want https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/resourceprocessor/README.md

**Reply by adriangb** [reply]
It looks like that can be used to rename, extract, etc. data from existing attributes. These attributes don't exist yet, I need to pull them from pod labels/annotations or such. Unless I'm misunderstanding what `resourceprocessor` can do?

**Reply by jrcamp** [reply]
If it's information you want to extract from k8s then yes you'd want to use `k8s_tagger`. See:

https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/k8sprocessor/testdata/config.yaml

**Reply by adriangb** [reply]
Got it, thank you for confirming.
